src/FastField.tsx

```diff
@@ -111,25 +112,44 @@ export class FastField<
         const maybePromise = (validate as any)(
           setIn(values, this.props.name, val)
         );
-        console.log('validate');
-        console.log(
-          diffObjectsExceptForKey(maybePromise, errors, this.props.name)
-        );
+
         if (isPromise(maybePromise)) {
           this.setState({ value: val });
           (maybePromise as any).then(
             () => this.setState({ error: undefined }),
-            (error: any) =>
-              this.setState({ error: getIn(error, this.props.name) })
+            (error: any) => {
+              // Here we diff the errors object relative to Formik parents except for
+              // the Field's key. If they are equal, the field's validation function is
+              // has no inter-field side-effects and we only need to update local state
+              // otherwise we need to lift up the update to the parent (causing a full form render)
+              if (isEqualExceptForKey(maybePromise, errors, this.props.name)) {
```

Comment:
This diffs the errors object between the parent's current errors and the one returned by validate. If it differs, then it means that the validation of the field has side effects. For example...

```js
const validate = (values: any) => {
  let errors: any = {};
  if (values.friends[0].name === values.email) {
    errors.email = 'cant be the same';
    errors.friends = [{ name: 'cant be the same' }];
  }
  return errors;
};```

In this situation, FastField will work act like the old Field and update the parent's Formik's state instead of the localized one--causing rerenders where necessary. This diff trick is not needed during onBlur / handleBlur because FastField _always_ updates the parent.